app/namespaces/test_adjustment/test_adjustment_service.py

```diff
@@ -0,0 +1,262 @@
+    # e.g., log odds within 1
+    logit_delta = logit(median_adj_prev) - logit(raw_prev)
+    adjusted_closeto_raw = abs(logit_delta) < 1
+
+    # permit the adjusted result if adjusted and raw are both < 0.5
+    # or if both are > 0.5
+    both_below_maxsmall = (median_adj_prev <= 0.5) and (raw_prev <= 0.5)
+    both_above_minbig = (median_adj_prev >= 0.5) and (raw_prev >= 0.5)
+
+    return (adjusted_closeto_raw or both_below_maxsmall or both_above_minbig)
+
+
+def validate_against_schema(input_payload, schema):
+    try:
+        payload = schema.load(input_payload)
+    except ValidationError as err:
+        return err.messages
+    return payload
+
+
+class ModelParamsSchema(Schema):
+    n_prev_obs = fields.Integer()
+    y_prev_obs = fields.Float()
+    n_se = fields.Integer()
+    y_se = fields.Float()
+    n_sp = fields.Integer()
+    y_sp = fields.Float()
+
+
+class TestAdjHandler:
+    def __init__(self, model_code=testadj_model_code, model_name='testadj_binomial_se_sp',
+                 execution_params={}):
+        self.TESTADJ_MODEL = self.get_stan_model_cache(model_code=model_code, model_name=model_name)
+        self.TESTADJ_MODEL_NAME = model_name
+        # Parse execution params
+        self.n_iter = execution_params.get('n_iter', 2000)
+        self.n_chains = execution_params.get('n_chains', 4)
+        self.return_fit = execution_params.get('return_fit', False)
+        self.n_replicates = execution_params.get('n_replicates', 5)
+        self.trials_lim = execution_params.get('trials_lim', 100)
+        self.modelsets_lim = execution_params.get('modelsets_lim', 3)
+
+    # make sure to gitignore model caches, because the model needs to be compiled separately
+    # on each machine - it is system-specific C++ code
+    def get_stan_model_cache(self, model_code: str, model_name: str = 'anon_model', **kwargs: Dict) -> pystan.StanModel:
+        """Use just as you would `pystan.StanModel`"""
+
+        # Create filepath of cached model
+        code_hash = md5(model_code.encode('ascii')).hexdigest()
+        cache_fn = f'stanmodelcache-{model_name}-{code_hash}.pkl'
+
+        # Try to load cached model
+        try:
+            cached_model = pickle.load(open(cache_fn, 'rb'))
+            print(f"Using cached StanModel at filepath {cache_fn}")
+
+        # Otherwise create the model and cache it
+        except FileNotFoundError:
+            cached_model = pystan.StanModel(model_code=model_code, model_name=model_name, **kwargs)
+            with open(cache_fn, 'wb') as f:
+                pickle.dump(cached_model, f)
+        return cached_model
+
+    def fit_one_pystan_model(self, model_params: Dict) -> Tuple:
+        fit = self.TESTADJ_MODEL.sampling(data=model_params,
+                                          iter=self.n_iter,
+                                          chains=self.n_chains,
+                                          control={'adapt_delta': 0.95},
```

Comment:
Shouldn't the hyperparameters here be different, @atatmaja @simonarocco? This is what Zayd was supposed to work on.

I think it was something like adapt_delta = 0.8, he set particular init values, etc... not sure that I see these reflected here

Reply:
@atatmaja any idea on this? not sure at all, never saw his work